src/datamap.rs

```diff
@@ -8,103 +8,82 @@ pub struct ChunkDetails {
+#[derive(PartialEq, Eq, PartialOrd, Ord)]
 pub enum DataMap {
-  /// if file was large enough (larger than 3072 bytes, 3 * MIN_CHUNK_SIZE) 
+  /// if file was large enough (larger than 3072 bytes, 3 * MIN_CHUNK_SIZE)
   /// this holds the list of chunks' info to decrypt it
   Chunks(Vec<ChunkDetails>),
-  /// very small files (less than 3072 bytes, 3 * MIN_CHUNK_SIZE) are put here in entirely 
+  /// very small files (less than 3072 bytes, 3 * MIN_CHUNK_SIZE) are put here in entirely
   Content(Vec<u8>),
   /// empty datamap
   None
-  }
+}
 
 impl DataMap {
   /// original size of file in datamap
-  pub fn len(&self)->u64 {
-    let size = 0u64;
+  pub fn len(&self) -> u64 {
     match *self {
-      DataMap::Chunks(ref chunks) => DataMap::chunks_size(chunks), 
-        DataMap::Content(ref content) => content.len() as u64, 
-        DataMap::None => 0u64
+      DataMap::Chunks(ref chunks) => DataMap::chunks_size(chunks),
+      DataMap::Content(ref content) => content.len() as u64,
+      DataMap::None => 0
     }
   }
+
   /// returning the list of chunk info if present
-  pub fn get_chunks(&self)->Vec<ChunkDetails> {
+  pub fn get_chunks(&self) -> Vec<ChunkDetails> {
     match *self {
-      DataMap::Chunks(ref chunks) => return chunks.to_vec(),
-        _                         => panic!("no chunks")
+      DataMap::Chunks(ref chunks) => chunks.to_vec(),
+      _ => panic!("no chunks")
     }
   }
 
-  /// we require this to be a sorted list to allow get_pad_iv_key to get the correct 
+  /// we require this to be a sorted list to allow get_pad_iv_key to get the correct
   /// pre encryption hashes for decrypt/encrypt
-  pub fn get_sorted_chunks(&self)->Vec<ChunkDetails> {
+  pub fn get_sorted_chunks(&self) -> Vec<ChunkDetails> {
     match *self {
       DataMap::Chunks(ref chunks) =>  {
-                                        let mut result = chunks.to_vec();
-                                        DataMap::chunks_sort(result.as_mut_slice());
-                                        result
-                                      }, 
-        _                           => panic!("no chunks")
+        let mut result = chunks.to_vec();
+        DataMap::chunks_sort(&mut result);
+        result
+      },
+      _ => panic!("no chunks")
     }
   }
 
   /// content stored as chunks or as raw
-  pub fn has_chunks(&self)->bool {
+  pub fn has_chunks(&self) -> bool {
     match *self {
-      DataMap::Chunks(ref chunks) => DataMap::chunks_size(chunks) > 0, 
-        _ => false, 
+      DataMap::Chunks(ref chunks) => DataMap::chunks_size(chunks) > 0,
+      _ => false,
     }
   }
 
   /// iterate through the chunks to figure out the total size, i.e. the file size
-  fn chunks_size(chunks: &Vec<ChunkDetails>)->u64 {
-    let mut size = 0u64;
-    for i in chunks.iter() {
-      size += i.source_size
-    }
-    return size
+  fn chunks_size(chunks: &[ChunkDetails]) -> u64 {
+    chunks.iter().fold(0, |acc, chunk| acc + chunk.source_size)
```

Comment:
Does Rust not need an explicit return statement? sorry for being an absolute Rust novice :)